File: backend/src/main/kotlin/com/etchedjournal/etched/models/entity/EncryptedEntity.kt
package com.etchedjournal.etched.models.entity

import com.etchedjournal.etched.models.OwnerType
import java.time.Instant
import javax.persistence.Column
import javax.persistence.Convert
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class EncryptedEntity(
    id: String,
    timestamp: Instant,
    owner: String,
    ownerType: OwnerType,

    @Column(nullable = false)
    val content: ByteArray,

    @Column(nullable = false, name = "key_pair_id")
    @Convert(converter = IdConverter::class)
    val keyPairId: String
) : BaseEntity(
    id = id,
    timestamp = timestamp,
    owner = owner,
    ownerType = ownerType
)

File: frontend-ng/src/app/services/models/encrypted-entity.ts
import { BaseEntity } from './base-entity';
import { Schema } from './schema';

export type Base64Str = string;

export interface EncryptedEntity extends BaseEntity {
    /** entity content as a base 64 encoded string */
    content: Base64Str;

    /** id of key pair used to encrypt this entity */
    keyPairId: string;

    schema: Schema;
}
